utils/wasm-gen/src/syscalls.rs

```diff
@@ -73,501 +132,173 @@ impl Default for SyscallsConfig {
     );
-    // gr_exit(value_dest_ptr: *const u8) -> !;
-    res.insert(
-        "gr_exit",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
-    );
-    // gr_gas_available() -> u64;
+    res.insert("gr_exit", SysCallInfo::new(config, [Ptr], [], frequency));
     res.insert(
         "gr_gas_available",
-        SysCallInfo {
-            params: [].to_vec(),
-            results: [I64].to_vec(),
-            param_rules: [].to_vec(),
-            frequency,
-        },
+        SysCallInfo::new(config, [], [I64], frequency),
     );
-    // gr_program_id(val: *mut u8);
     res.insert(
         "gr_program_id",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
+        SysCallInfo::new(config, [Ptr], [], frequency),
     );
-    // gr_origin(origin_ptr: *mut u8);
-    res.insert(
-        "gr_origin",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
-    );
-    // gr_leave() -> !;
-    res.insert(
-        "gr_leave",
-        SysCallInfo {
-            params: [].to_vec(),
-            results: [].to_vec(),
-            param_rules: [].to_vec(),
-            frequency,
-        },
-    );
-    // gr_value_available(val: *mut u8);
+    res.insert("gr_origin", SysCallInfo::new(config, [Ptr], [], frequency));
+    res.insert("gr_leave", SysCallInfo::new(config, [], [], frequency));
     res.insert(
         "gr_value_available",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
-    );
-    // gr_wait() -> !;
-    res.insert(
-        "gr_wait",
-        SysCallInfo {
-            params: [].to_vec(),
-            results: [].to_vec(),
-            param_rules: [].to_vec(),
-            frequency,
-        },
+        SysCallInfo::new(config, [Ptr], [], frequency),
     );
-    // gr_wait_up_to(duration: u32) -> !;
+    res.insert("gr_wait", SysCallInfo::new(config, [], [], frequency));
     res.insert(
         "gr_wait_up_to",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
+        SysCallInfo::new(config, [Size], [], frequency),
     );
-    // gr_wait_for(duration: u32) -> !;
     res.insert(
         "gr_wait_for",
-        SysCallInfo {
-            params: [I32].to_vec(),
-            results: [].to_vec(),
-            param_rules: [ptr_rule()].to_vec(),
-            frequency,
-        },
+        SysCallInfo::new(config, [Delay], [], frequency),
```

Comment:
That's not delay, that's duration

Reply:
Is there difference between `delay` and `duration` in syscalls ?
```
       pub fn gr_wait_for(duration: u32) -> !;

       pub fn gr_wake(message_id_ptr: *const [u8; 32], delay: u32) -> SyscallError;
```

Reply:
`gr_wait_for` and `gr_wake` is totally different things.

Delay for sending messages and `gr_wake` means amount of blocks system should wait before do requested stuff.

Duration allowed for `gr_wait`-like syscalls and means how many blocks message should be in waitlist

Reply:
Done